sdk/executor/src/main/java/com/mesosphere/sdk/executor/ProcessTask.java

```diff
@@ -62,6 +62,19 @@ protected ProcessTask(
         this.taskInfo = taskInfo;
         this.processBuilder = processBuilder;
         this.exitOnTermination = exitOnTermination;
+        this.taskKillGracePeriod = getTaskKillGracePeriod(taskInfo);
+    }
+
+    private Duration getTaskKillGracePeriod(Protos.TaskInfo taskInfo) {
```

Comment:
Please remove `null` checks and use protobuf `hasKillPolicy()` and `hasGracePeriod()` style instead.